### PR TITLE
Update sdk for aptos wallet service

### PR DESCRIPTION
**Pull request by jachen-sh**

### CODE REVIEW

File: packages/clients/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@shinami/clients",
-  "version": "0.9.0",
+  "version": "0.10.0",
```
Comment (jasonxh) [inline packages/clients/package.json:3]: Let's do `0.9.1` since it's non-breaking.

File: packages/clients/package.json [hunk 2]
```diff
@@ -64,7 +64,7 @@
     "superstruct": "^1.0.3"
   },
   "devDependencies": {
-    "@aptos-labs/ts-sdk": ">=1.13.3",
+    "@aptos-labs/ts-sdk": ">=1.22.2",
```
Comment (jasonxh) [inline packages/clients/package.json:67]: Are we using any new features from this version?

File: packages/clients/test/aptos/wallet.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  AccountAuthenticator,
+  AccountAuthenticatorEd25519,
+  isUserTransactionResponse,
+} from "@aptos-labs/ts-sdk";
+
+const aptos = createAptos();
+const key = createKeyClient();
+const wal = createWalletClient();
+
+function delay(ms: number) {
```
Comment (jasonxh) [inline packages/clients/test/aptos/wallet.integration.test.ts:27]: I usually avoid this in favor of more targeted waits. Is there a wait helper from Aptos sdk we can use or some other thing we can poll for?

File: packages/nextjs-zklogin/package.json [hunk 4]
```diff
@@ -65,7 +65,7 @@
   },
   "dependencies": {
     "@mysten/zklogin": "^0.7.4",
-    "@shinami/clients": "^0.9.0",
+    "@shinami/clients": "^0.10.0",
```
Comment (jasonxh) [inline packages/nextjs-zklogin/package.json:68]: No need to bump this since it's not using the new code in here.

File: packages/clients/src/aptos/wallet.ts [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+        return (await this.walletClient.getWallet(this.walletId))
+          .accountAddress;
+      }
+      throw e;
+    }
+  }
+
+  /**
+   * Tries to create this wallet if it doesn't exist.
+   * @param onChain If set, it will try to create the wallet and initialize it on chain as well.
```
Comment (jasonxh) [inline packages/clients/src/aptos/wallet.ts:351]: Clarify this must be called with an access key with gas auth, and we'll draw from the fund.

File: packages/clients/src/aptos/wallet.ts [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+  createSession(secret: string): Promise<string> {
+    return this.request("key_createSession", [secret], string());
+  }
+}
+
+/**
+ * Data wrapper for a Shinami Aptos wallet.
+ */
+const WalletInfo = object({
+  accountAddress: string(),
```
Comment (jasonxh) [inline packages/clients/src/aptos/wallet.ts:47]: There has been a design philosophy shift in Aptos gas client that we are integrating more closely with Aptos sdk using native types. I suggest making this `AccountAddress` because it's returned to callers (you should also `export` the type), and you can create a sister type `GetWalletResult` for internal JSON RPC typing.

File: packages/clients/src/aptos/wallet.ts [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+   * @param walletId Wallet id.
+   * @param sessionToken Session token, obtained by `KeyClient.createSession`.
+   * @param transaction Transaction. Can be SimpleTransaction or MultiAgentTransaction
+   * @returns Signing Result
+   */
+  async signTransaction(
+    walletId: string,
+    sessionToken: string,
+    transaction: AnyRawTransaction,
+  ): Promise<SignTransactionResult> {
```
Comment (jasonxh) [inline packages/clients/src/aptos/wallet.ts:152]: Along the same line as above, I suggest returning `AccountAuthenticator` here.

File: packages/clients/package.json (hunk 1 above)
Comment (jachen-sh) [reply packages/clients/package.json:3]: Would this not qualify as a minor version bump? Adding functionality in backwards compatible (non breaking) manner: https://semver.org/#semantic-versioning-200

File: packages/clients/package.json (hunk 2 above)
Comment (jachen-sh) [reply packages/clients/package.json:67]: No new features, but wanted to keep versioning up to date. Do you recommend keeping the previous version?

File: packages/clients/package.json (hunk 1 above)
Comment (jasonxh) [reply packages/clients/package.json:3]: Normally yes, new features without breaking chagnes should be introduced with minor version bumps. The special case here is we are still on `0.*`, in which case even minor version bumps are considered breaking by [npm](https://semver.npmjs.com/). E.g.  `0.10.0` is NOT compatible with `^0.9.0`.

File: packages/clients/package.json (hunk 2 above)
Comment (jasonxh) [reply packages/clients/package.json:67]: Here it's not about "keeping it up to date". It's specifying a minimum bar for the [peer dependency](https://nodejs.org/en/blog/npm/peer-dependencies). The lower the bar, the more compatible this library is in dependent projects.

> One piece of advice: peer dependency requirements, unlike those for regular dependencies, should be lenient. You should not lock your peer dependencies down to specific patch versions. It would be really annoying if one Chai plugin peer-depended on Chai 1.4.1, while another depended on Chai 1.5.0, simply because the authors were lazy and didn't spend the time figuring out the actual minimum version of Chai they are compatible with.

File: packages/clients/src/aptos/wallet.ts [hunk 8]
```diff
@@ -0,0 +1,431 @@
...
+   * Creates a new wallet that is not initialized on chain.
+   * @param walletId Wallet id. Must not have been previously used, or an error will be returned.
+   * @param sessionToken Session token. Obtained by `KeyClient.createSession`.
+   * @returns Aptos address of the created wallet.
+   */
+  async createWallet(
+    walletId: string,
+    sessionToken: string,
+  ): Promise<WalletInfo> {
+    return this.request(
```
Comment (aaronTufts) [inline packages/clients/src/aptos/wallet.ts:91]: For this and the other methods that return an AccountAddress, it would be nice if we unpack it and return a string, like we do with ShinamiWalletSigner (and SUI, though the difference there starts with the Sui Wallet Service returning a string and not an object). There may be something I'm overlooking, but for user convenience and for consistency, it would be nice to just return the string.

Comment (jachen-sh) [reply packages/clients/src/aptos/wallet.ts:91]: Yea this is the same as @jasonxh 's comments. I'll be making the change to unpack it directly to the aptos sdk AccountAddress type (rather than just a string)

File: packages/clients/test/aptos/wallet.integration.test.ts (hunk 3 above)
Comment (jachen-sh) [reply packages/clients/test/aptos/wallet.integration.test.ts:26]: Couldn't find anything specific in aptos sdk, opted to poll on fetching account info in a retry loop

File: packages/clients/src/aptos/wallet.ts (hunk 6 above)
Comment (jachen-sh) [reply packages/clients/src/aptos/wallet.ts:49]: Made this more in line with returning native Aptos sdk types. One small deviation from your suggestion: I used `WalletResult` as the internal json rpc type since it's used in the wallet operation methods (create, initialize), not just `getWallet`

Comment (jasonxh) [reply packages/clients/src/aptos/wallet.ts:49]: For consistency, name this `GetWalletResult`.

File: packages/clients/README.md [hunk 9]
```diff
@@ -313,6 +314,130 @@ const committed = await aptos.transaction.waitForTransaction({
...
+const wal = new WalletClient(WALLET_ACCESS_KEY);
+
+// WALLET_SECRET MUST be used consistently with this wallet id.
+// You are responsible for safe-keeping the (walletId, secret) pair.
+// Shinami cannot recover it for you.
+const signer = new ShinamiWalletSigner("my_wallet_id", wal, WALLET_SECRET, key);
+
+// Get or create the sender account on chain. Since this is a non-sponsored transaction,
+// your account MUST have APT in it in order to execute a transaction.
+const senderAccount = AccountAddress.from(await signer.getAddress(true, true));
```
Comment (jasonxh) [inline packages/clients/README.md:350]: Update

File: packages/clients/README.md [hunk 10]
```diff
@@ -313,6 +314,130 @@ const committed = await aptos.transaction.waitForTransaction({
...
+  options: {
+    expireTimestamp: Math.floor(Date.now() / 1000) + 5 * 60,
+  },
+});
+
+// Sign tx with invisible wallet.
+const { signature } = await signer.signTransaction(transaction);
+
+// Deserialize the result to an AccountAuthenticator
+const accountAuthenticator = AccountAuthenticator.deserialize(
```
Comment (jasonxh) [inline packages/clients/README.md:369]: Update

File: packages/clients/README.md [hunk 11]
```diff
@@ -313,6 +314,130 @@ const committed = await aptos.transaction.waitForTransaction({
...
+const wal = new WalletClient(SUPER_ACCESS_KEY);
+
+// WALLET_SECRET MUST be used consistently with this wallet id.
+// You are responsible for safe-keeping the (walletId, secret) pair.
+// Shinami cannot recover it for you.
+const signer = new ShinamiWalletSigner("my_wallet_id", wal, WALLET_SECRET, key);
+
+// Get or create the sender account on chain. You do not need APT in this account since
+// we'll be using Shinami's gas station to sponsor the transaction.
+const senderAccount = AccountAddress.from(await signer.getAddress(true, true));
```
Comment (jasonxh) [inline packages/clients/README.md:417]: Update

File: packages/clients/src/aptos/wallet.ts (hunk 8 above)
Comment (jasonxh) [reply packages/clients/src/aptos/wallet.ts]: I think @aaronTufts was asking for the exact opposite, which I don't agree with. The new design we are going with is to integrate as seamlessly with the rest of the SDK as possible, so that using a `ShinamiWalletSigner` feels like using a builtin signer without unnecessary pre or post processing. The difference can be easily seen from the updated examples. If the user really needs a string, Aptos SDK already makes it very easy, and I assume the user is already familiar with the steps because they use Aptos SDK way more than Shinami SDK.

File: package-lock.json
Comment (jasonxh) [inline package-lock.json:1]: Revert the unnecessary changes here.

File: packages/clients/src/aptos/wallet.ts (hunk 8 above)
Comment (aaronTufts) [reply packages/clients/src/aptos/wallet.ts]: For clarity: I prefer where we ended up.

When I said "AccountAddress" I misspoke and did not mean to refer to the Aptos type, but instead to the string we were returning, which I believe was called "accountAddress". I meant to ask for consistency, not a string vs something else. Sorry for the confusion from lack of precision.

File: packages/clients/src/aptos/wallet.ts (hunk 6 above)
Comment (jachen-sh) [reply packages/clients/src/aptos/wallet.ts:49]: resolved offline. using `WalletResult`

**Review by jasonxh — APPROVED** [review]
